tests/test_minimal/test_utils/test_get_json_schema_from_method_signature.py

```diff
@@ -0,0 +1,314 @@
+from pathlib import Path
+from typing import Dict, List, Literal, Optional, Union
+
+import pytest
+from jsonschema import validate
+from pydantic import DirectoryPath, FilePath
+
+from neuroconv.datainterfaces import AlphaOmegaRecordingInterface
+from neuroconv.utils import ArrayType, get_json_schema_from_method_signature
+
+
+def test_get_json_schema_from_method_signature_basic():
+    def basic_method(
+        integer: int,
+        floating: float,
+        string_or_path: Union[Path, str],
+        boolean: bool,
+        literal: Literal["a", "b", "c"],
+        dictionary: Dict[str, str],
+        string_with_default: str = "hi",
+        optional_dictionary: Optional[Dict[str, str]] = None,
+    ):
+        pass
+
+    test_json_schema = get_json_schema_from_method_signature(method=basic_method)
+    expected_json_schema = {
+        "additionalProperties": False,
+        "properties": {
+            "boolean": {"type": "boolean"},
+            "dictionary": {"additionalProperties": {"type": "string"}, "type": "object"},
+            "floating": {"type": "number"},
+            "integer": {"type": "integer"},
+            "literal": {"enum": ["a", "b", "c"], "type": "string"},
+            "optional_dictionary": {
+                "anyOf": [{"additionalProperties": {"type": "string"}, "type": "object"}, {"type": "null"}],
+                "default": None,
+            },
+            "string_or_path": {"anyOf": [{"format": "path", "type": "string"}, {"type": "string"}]},
+            "string_with_default": {"default": "hi", "type": "string"},
+        },
+        "required": ["integer", "floating", "string_or_path", "boolean", "literal", "dictionary"],
+        "type": "object",
+    }
+
+    assert test_json_schema == expected_json_schema
+
+
+def test_get_json_schema_from_method_signature_advanced():
+    """
+    These are annotations newly supported by the Pydantic-based inference.
+
+    They should also be compatible with __future__.annotations for SpikeInterface.
+    """
+
+    # TODO: enable | instead of union when 3.11 is minimal
+    def advanced_method(
+        old_list_of_strings: List[str],
+        new_list_of_strings: list[str],
+        old_dict_of_ints: Dict[str, int],
+        new_dict_of_ints: dict[str, int],
+        nested_list_of_strings: List[List[str]],
+        array_type: ArrayType,
```

Comment:
@h-mayorquin Here is an explicit test of array

Previously I was simply relying on their presence in various data interfaces, which failed their respective data interface tests without the config fix

Reply:
Note however that it is (must be) via our usage of the `ArrayType` (which itself could be improved no doubt); if it is simply a `np.array` only then it must be excluded in order to pass this inference

Reply:
OK, I just tested using the official numpy typing in your test:

```python
    from numpy.typing import ArrayLike
    # TODO: enable | instead of union when 3.11 is minimal
    def advanced_method(
        old_list_of_strings: List[str],
        new_list_of_strings: list[str],
        old_dict_of_ints: Dict[str, int],
        new_dict_of_ints: dict[str, int],
        nested_list_of_strings: List[List[str]],
        array_type: ArrayLike,
        # more_nested_list_of_strings: list[list[list[str]]],
        # pathalogical_case: list[dict[str | int | None, list[Optional[dict[str, list[Literal["a", "b"] | None]]]]],],
    ):
        pass

    test_json_schema = get_json_schema_from_method_signature(method=advanced_method)
```

The schema returns
```python
test_json_schema["properties"]["array_type"]
{'anyOf': [{'type': 'boolean'}, {'type': 'integer'}, {'type': 'number'}, {'type': 'string'}, {'format': 'binary', 'type': 'string'}]}
```

But this is not yet supported:
https://github.com/pydantic/pydantic-extra-types/issues/31

Something to keep an eye on.

Reply:
Yes, exactly. People have been working on this for a long time and I haven't seen a single perfect accepted solution yet